tests/DocumentationExamplesTest.php

```diff
@@ -1751,4 +1867,39 @@ private function assertInventoryCount($count): void
     {
         $this->assertCollectionCount($this->getDatabaseName() . '.' . $this->getCollectionName(), $count);
     }
+
+    private function waitForSnapshot(string $databaseName, string $collectionName): void
```

Comment:
Note: this function was lifted from https://github.com/mongodb/mongo-ruby-driver/commit/7c4117b58e3d12e237f7536f7521e18fc15f79ac#diff-7b02c2b4b34a58c688750366d0230014bb453e0381af394d736aa9d0a25538a5R374 in the Ruby driver.

The function itself is not generally useful, since we're targeting a specific collection. The server-side error message is:

> Unable to read from a snapshot due to pending collection catalog changes; please retry the operation.

If a user were to encounter this, they would probably want to retry their original query. For our purposes here, ensuring _any_ snapshot query (even one with a different query filter) succeeds seems to do the trick to avoid an error in the documentation example that calls this method.